core/src/main/scala/org/locationtech/rasterframes/extensions/RasterJoin.scala

```diff
@@ -0,0 +1,124 @@
+  }
+
+  // NB: Don't be tempted to make this a `val`. Spark will barf if `withRasterFrames` hasn't been called first.
+  def reproject_and_merge = udf(reproject_and_merge_f).withName("reproject_and_merge")
+
+  def apply(left: DataFrame, right: DataFrame): DataFrame = {
+    val df = apply(left, right, left("extent"), left("crs"), right("extent"), right("crs"))
+    df.drop(right("extent")).drop(right("crs"))
+  }
+
+  def apply(left: DataFrame, right: DataFrame, leftExtent: Column, leftCRS: Column, rightExtent: Column, rightCRS: Column): DataFrame = {
+    val leftGeom = st_geometry(leftExtent)
+    val rightGeomReproj = st_reproject(st_geometry(rightExtent), rightCRS, leftCRS)
+    val joinExpr = st_intersects(leftGeom, rightGeomReproj)
+    apply(left, right, joinExpr, leftExtent, leftCRS, rightExtent, rightCRS)
+  }
+
+  def apply(left: DataFrame, right: DataFrame, joinExprs: Column, leftExtent: Column, leftCRS: Column, rightExtent: Column, rightCRS: Column): DataFrame = {
+    // Unique id for temporary columns
+    val id = Random.alphanumeric.take(5).mkString("_", "", "_")
+
+    // Post aggregation left extent. We preserve the original name.
+    val leftExtent2 = leftExtent.columnName
+    // Post aggregation left crs. We preserve the original name.
+    val leftCRS2 = leftCRS.columnName
+    // Post aggregation right extent. We create a new name.
+    val rightExtent2 = id + "extent"
+    // Post aggregation right crs. We create a new name.
+    val rightCRS2 = id + "crs"
+
+    // A representative tile from the left
+    val leftTile = left.tileColumns.headOption.getOrElse(throw new IllegalArgumentException("Need at least one target tile on LHS"))
+
+    // Gathering up various expressions we'll use to construct the result.
+    // After joining We will be doing a groupBy the LHS. We have to define the aggregations to perform after the groupBy.
+    // On the LHS we just want the first thing (subsequent ones should be identical.
+    val leftAggCols = left.columns.map(s => first(left(s), true) as s)
+    // On the RHS we collect result as a list.
+    val rightAggCtx = Seq(collect_list(rightExtent) as rightExtent2, collect_list(rightCRS) as rightCRS2)
+    val rightAggTiles = right.tileColumns.map(c => collect_list(c) as c.columnName)
+    val aggCols = leftAggCols ++ rightAggTiles ++ rightAggCtx
+
+    // After the aggregation we take all the tiles we've collected and resample + merge into LHS extent/CRS.
+    val reprojCols = rightAggTiles.map(t => reproject_and_merge(
+      col(leftExtent2), col(leftCRS2), col(t.columnName), col(rightExtent2), col(rightCRS2), rf_dimensions(leftTile)
+    ) as t.columnName)
+    val finalCols = leftAggCols.map(c => col(c.columnName)) ++ reprojCols
+
+    // Here's the meat:
+    left
+      // 1. Add a unique ID to each LHS row for subequent grouping.
+      .withColumn(id, monotonically_increasing_id())
+      // 2. Perform the left-outer join
+      .join(right, joinExprs, joinType = "left")
```

Comment:
I believe this requires spark conf `spark.sql.crossJoin.enabled=true` .... should we try to warn the user ahead of time, or should we enable that when doing the withRasterFrames spark session goodies?

Reply:
My understanding was that this should only be required if we were joining a DF on itself or doing an outer join with no predicate.... I had enabled it in the `TestEnvironment`, but we know how that worked out 🙄